monkey/infection_monkey/utils/signal_handler.py

```diff
@@ -0,0 +1,30 @@
+import logging
+import signal
+
+from infection_monkey.i_master import IMaster
+from infection_monkey.utils.environment import is_windows_os
+from infection_monkey.utils.exceptions.planned_shutdown_exception import PlannedShutdownException
+
+logger = logging.getLogger(__name__)
+
+
+class StopSignalHandler:
+    def __init__(self, master: IMaster):
+        self._master = master
+
+    def __call__(self, signum, __=None):
```

Comment:
We should be wary of using the `__call__`. It's meant to hide the state from the caller, which is unnecessary in this case. Generally, it's better to have
```
signal.signal(signal.SIGINT, stop_signal_handler.handle_signal)
```
because this way it's clear that `signal.signal` expects to get a function. As it is right now, it looks like it expects an object of class `StopSignalHandler` until you actually read the `StopSignalHandler` or `signal.signal`